src/patternfly/components/TreeView/tree-view.scss

```diff
@@ -0,0 +1,196 @@
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-1__list-item--m-expandable__link--PaddingLeft);
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-1__list-item__link--PaddingLeft);
+      }
+    }
+  }
+
+  &.pf-m-nested-level-2 {
+    .pf-c-tree-view__list-item {
+      &.pf-m-expandable > .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-2__list-item--m-expandable__link--PaddingLeft);
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-2__list-item__link--PaddingLeft);
+      }
+    }
+  }
+
+  &.pf-m-nested-level-3 {
+    .pf-c-tree-view__list-item {
+      &.pf-m-expandable > .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-3__list-item--m-expandable__link--PaddingLeft);
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-3__list-item__link--PaddingLeft);
+      }
+    }
+  }
+}
+// stylelint-enable
+
+.pf-c-tree-view__list-item {
+  &.pf-m-expanded {
+    > .pf-c-tree-view__link > .pf-c-tree-view__link-toggle {
+      transform: rotate(var(--pf-c-tree-view__link--m-expanded__toggle--Rotate));
+    }
+  }
+}
+
+.pf-c-tree-view__link {
+  display: flex;
+  align-items: center;
+  width: 100%;
+  padding: var(--pf-c-tree-view__link--PaddingTop) var(--pf-c-tree-view__link--PaddingRight) var(--pf-c-tree-view__link--PaddingBottom) var(--pf-c-tree-view__link--PaddingLeft);
+  color: var(--pf-c-tree-view__link--Color);
+  text-align: left;
+  border: 0;
+
+  &.pf-m-current {
+    --pf-c-tree-view__link--Color: var(--pf-c-tree-view__link--m-current--Color);
+
+    font-weight: var(--pf-c-tree-view__link--m-current--FontWeight);
+  }
+
+  &.pf-m-first {
+    font-weight: var(--pf-c-tree-view__link--m-first--FontWeight);
+  }
+
+  &:hover {
+    background-color: var(--pf-c-tree-view__link--hover--BackgroundColor);
+  }
+
+  .pf-c-check {
+    --pf-c-check__input--MarginTop: 0;
+    --pf-c-check--GridGap: 0;
+
+    padding-right: var(--pf-c-tree-view__link--c-check--PaddingRight);
+  }
+
+  .pf-c-badge {
+    margin-left: var(--pf-c-tree-view__link--c-badge--MarginLeft);
+
+    &.pf-m-read {
+      --pf-c-badge--m-read--BackgroundColor: var(--pf-c-tree-view__link--c-badge--BackgroundColor);
+    }
+  }
+}
+
+.pf-c-tree-view__link-toggle {
+  display: inline-flex;
+  align-items: center;
+  justify-content: center;
+  min-width: var(--pf-c-tree-view__link-toggle--MinWidth);
+  min-height: var(--pf-c-tree-view__link-toggle--MinHeight);
+  padding-right: var(--pf-c-tree-view__link-toggle--PaddingRight);
+  padding-left: var(--pf-c-tree-view__link-toggle--PaddingLeft);
+}
+
+.pf-c-tree-view__link-text {
+  overflow: hidden;
```

Comment:
wanna use `@include pf-text-overflow;` here?

Reply:
Actually you could use the line-clamp mixin that would allow a user to specify they want to truncate after the value of an optional `--pf-c-tree-view__link-text--max-lines` var, and there could be a prop in react for that.

<img width="323" alt="Screen Shot 2020-07-20 at 6 44 35 PM" src="https://user-images.githubusercontent.com/35148959/87996925-42669100-cab9-11ea-8b9d-47e4ed5b883c.png">


https://github.com/patternfly/patternfly/blob/3ab70e9587c143ec2ff06e1b99b0cb2cefd07305/src/patternfly/sass-utilities/mixins.scss#L153-L160

Reply:
also you'll want to add `word-break: break-word;` regardless

Reply:
Tried doing line-clamp vs text-overflow and the spacing to the left is larger for line-clamp, do you know why this might be 

<img width="694" alt="Screen Shot 2020-07-24 at 3 05 34 PM" src="https://user-images.githubusercontent.com/20118816/88427331-b4dcb700-cdc0-11ea-9a0f-cedaabf27190.png">

<img width="688" alt="Screen Shot 2020-07-24 at 3 06 06 PM" src="https://user-images.githubusercontent.com/20118816/88427333-b60de400-cdc0-11ea-8b4b-13c0f55d54d8.png">

Reply:
are you referring to the space after the `...`? If so it's because `line-clamp` will actually lay something out in as many lines as it would be naturally, and clip the end of the text on whatever line you specify with `line-clamp`.  If you changed `line-clamp` to "2", you'll see that the text doesn't shift. Here's an exaggerated example to show that https://codepen.io/mcoker/pen/gOPEqge

Reply:
@mcoker got it. So if there are multiple lines that overflow, the differences could look like this. If the different spaces on the right seems natural to you, im ok with it

<img width="676" alt="Screen Shot 2020-07-27 at 9 48 11 AM" src="https://user-images.githubusercontent.com/20118816/88549536-690f5500-cfee-11ea-9e28-5670f9e8f22a.png">

Reply:
Good point, using `@include pf-text-overflow;` is fine. If users want to specify the number of lines, we can consider the other way with this caveat.